### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by ceceliahavens**
I have one test that needs a little help, but after that I think it is ready to merge.  Delete, and edit capabilities are in place as well as the TDD for the comments.  Please send me some feedback on the TDD and I will fix the one test that is commented out before we merge into the master.

### CODE REVIEW

File: test/controllers/comments_controller_test.rb [hunk 1]
```diff
@@ -1,7 +1,45 @@
 require 'test_helper'
 
 class CommentsControllerTest < ActionController::TestCase
-  # test "the truth" do
-  #   assert true
+  test "create comment requires logged in" do 
+    post :create, :post_id => 'Tacocats'
```
Comment (kenmazaika) [inline test/controllers/comments_controller_test.rb:5]: nice!

File: test/controllers/comments_controller_test.rb [hunk 2]
```diff
@@ -1,7 +1,45 @@
...
+    comment = Comment.last
+    assert_equal u, comment.user
+  end
+  
+  # test "create comment needs a valid post_id" do
+  #   u = FactoryGirl.create(:user)
+  #   sign_in u
+  #   post :create, :post_id => 'omg'
+  #   assert_response :not_found
   # end
```
Comment (kenmazaika) [inline test/controllers/comments_controller_test.rb:28]: This test looks spot on to me.  What error message do you get?

**Comment by kenmazaika** [comment]
Looks good to me!  Could use a few more tests for edit, update and destroy, but overall looks good.  Happy to merge this if you're happy with it.

File: test/controllers/comments_controller_test.rb (hunk 2 above)
Comment (ceceliahavens) [inline test/controllers/comments_controller_test.rb:28]: 1) Error:
CommentsControllerTest#test_create_comment_needs_a_valid_post_id:
ActiveRecord::RecordNotFound: Couldn't find Post with id=omg
    app/controllers/comments_controller.rb:5:in `create'
    test/controllers/comments_controller_test.rb:26:in`block in class:CommentsControllerTest'

**Comment by ceceliahavens** [comment]
Let's merge this one  but not close the issue, and I will work on those tests tomorrow.

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -1,13 +1,42 @@
 class CommentsController < ApplicationController
   before_action :authenticate_user!
+
 def create
   @post = Post.find(params[:post_id])
```
Comment (kenmazaika) [inline app/controllers/comments_controller.rb:5]: This line is causing problems in your tests.  Right now you have:

```
   @post = Post.find(params[:post_id])
```

Which raises an error if something isn't found.  Instead we could do something like:

```
@post = Post.find_by_id(params[:post_id])
render_not_found if @post.blank?
```

To fix it.

**Comment by kenmazaika** [comment]
Works for me! :) :+1:  Great work.

**Pull request merged**